.github/workflows/clang-format.yml

```diff
@@ -0,0 +1,35 @@
+name: clang-format
+
+# Report style formatting changes.
+# Run git-clang-format on the contents of the PR against the target commit and display a diff of
+# any recommended changes.  Fail the job if any are recommended.
+#
+# Future work would be to have a way to apply these automatically which would require making and
+# pushing a new git commit so should be optional.
+#
+# Runs in a fedora docker container to get a recent clang-format version.
+
+on:
+  pull_request:
+
+jobs:
+  pylint:
+    name: Clang Format
+    runs-on: ubuntu-22.04
```

Comment:
Curious. How does this relate to `FROM fedora:38` from the docker file?
The action runs on ubuntu but the actual check on fedora?

Reply:
It runs in GitHub actions, the options are ubuntu of various versions, mac os or Windows, this check runs on ubuntu but uses a fedora Docker container to do the heavy lifting - the clang-format versions are too old on either ubuntu or el8.